watertap/unit_models/crystallizer.py

```diff
@@ -671,12 +670,11 @@ def initialize(
         init_log.info_high("Initialization Step 3 {}.".format(idaeslog.condition(res)))
         # ---------------------------------------------------------------------
         # Release Inlet state
-        blk.properties_in.release_state(flags, outlvl=outlvl)
-        init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
-
-        if not check_optimal_termination(res):
-            if raise_on_failure:
-                raise InitializationError(f"Unit model {blk.name} failed to initialize")
+        try:
+            blk.properties_in.release_state(flags, outlvl=outlvl)
+            init_log.info("Initialization Complete: {}".format(idaeslog.condition(res)))
+        except InitializationError:
+            raise InitializationError(f"Unit model {blk.name} failed to initialize")
```

Comment:
I'm not sure that we want the try/except here. I figured the user would wrap their `initialize() `step in a try/except to keep moving if an `InitializationError` were to occur.

The way this is implemented now, I think the `InitalizationError` might actually get skipped.

Reply:
I agree with Adam -- we should be checking the result of the `solve` method call do know whether or not to raise a `InitializationError`.

We should always release the state of the associated properties so that if the `InitializationError` is caught the unit has no more fixed variables than it did before the call to initialize.

Reply:
I will echo @bknueven's comment, and emphasize that if possible you should do the check after the release state step.

```
res = solver.solver(model)

release_state(model)

if not check_optimal_termination(model):
    raise InitializationError()
```

This way, the model should be left in its original state, allowing you to try to continue on (if desired). If you don't do this and try to ignore the error and solve the flowsheet, the problem will be over-specified due to the fixed inlet states.